### PR TITLE
Add support for second level cache

### PR DESCRIPTION
**Pull request by bakura10**
Addresses #295

Tests need to be written.

### CODE REVIEW

**Comment by bakura10** [comment]
Tests and doc have been added @Ocramius. That's now mergeable :).

**Comment by bakura10** [comment]
Again, tests are failing on PHP 5.3 only because of the minimum dependency required in Doctrine 2.5.

**Comment by stof** [comment]
@Ocramius why not using a version check to be compatible with both 2.4 and 2.5, as done in DoctrineBundle ? Forcing ZF users to wait until 2.5 is stable before trying the feature is bad. We want them to be able to test features during the beta and RC phases (otherwise these phases are less useful)

**Comment by Ocramius** [comment]
> why not using a version check

Could be done, I just don't have time to deal with it myself right now.

**Comment by stof** [comment]
@bakura10 can you add a version check in your code to allow keeping the compatibility with the ORM 2.4 ?

**Comment by bakura10** [comment]
Done @stof 

@Ocramius , I think we can merge this and #374 , then upgrading minimum PHP version to PHP 5.4 immediately?

File: composer.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
     "require":           {
         "php":                               ">=5.3.23",
         "doctrine/doctrine-module":          "0.9.*",
-        "doctrine/orm":                      ">=2.4,<2.6-dev",
-        "doctrine/dbal":                     ">=2.4,<2.6-dev",
+        "doctrine/orm":                      ">=2.5,<2.7-dev",
+        "doctrine/dbal":                     ">=2.5,<2.7-dev",
```
Comment (stof) [inline composer.json]: The goal of adding a version check is to preserve the compatibility with 2.4 too

**Comment by bakura10** [comment]
Done @stof . @Ocramius , ready to merge?

**Comment by Ocramius** [comment]
@bakura10 needs rebase

**Comment by bakura10** [comment]
This is mergeable @Ocramius !

**Pull request merged**